src/SelfDestructLinkButton/SelfDestructLinkButton.js

```diff
@@ -0,0 +1,95 @@
+// Copyright (c) Cosmo Tech.
+// Licensed under the MIT license.
+
+import React, { useState, useRef } from 'react';
+import PropTypes from 'prop-types';
+import {
+  Button,
+  CircularProgress
+} from '@material-ui/core';
+import useStyles from './style';
+
+const STATUS = {
+  IDLE: 'IDLE',
+  GENERATING: 'GENERATING',
+  READY: 'READY'
+};
+
+const defaultLabels = {
```

Comment:
You can put this directly in defaultProps. It will avoid you to check if labels is specify or not in following lines

Reply:
I'm worried about users providing an empty dict for labels: if it replaces the default values, nothing will be displayed.
Another solution would be to merge the labels prop with the default labels dict:
```
const mergedLabels = {
  ...defaultLabels,
  ...labels
}
```

Reply:
After discussion, default labels will be set directly in `defaultProps` to stay consistent with the current refactoring for components documentation :+1: